sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -451,13 +451,17 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
     foreachImplementation(writer.asInstanceOf[ForeachWriter[Any]])
   }
 
-  private[sql] def foreachImplementation(writer: ForeachWriter[Any]): DataStreamWriter[T] = {
+  private[sql] def foreachImplementation(writer: ForeachWriter[Any],
+      encoder: Either[ExpressionEncoder[Any], InternalRow => Any] = null): DataStreamWriter[T] = {
```

Comment:
`Either` implementation is not needed. Only `Left` is used. 
Actually this change is not needed at all, I think.

Reply:
Yes, we are not supporting custom encoder yet. The unit test is disabled.

Reply:
We can make this change when we have that working.

Reply:
I think we need to be cautious here. Would it be a problem if we update this in the future. And a client with old code try to connect to the new server?

Reply:
But we don't know if this is correct.

Reply:
Also I think even with only generic types the encoder is still needed. Let me test it out

Reply:
Confirmed, I make the encoder to be null and run the `int` code
```
Caused by: java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema cannot be cast to java.lang.Integer
```

This is because on the server, the ds is always df (Dataset[Row]), and therefore ds.expenc doesn't cast the row to Int, which the `process` method uses